app/views/shared/_error_messages.html.erb

```diff
@@ -0,0 +1,13 @@
+<% if @user.errors.any? %>
+  <div id="error_explanation">
+  <div class="alert alert-danger">
+  <%= t "the_form_contains" %>
+  <%= pluralize @user.errors.count, "error" %>.
```

Comment:
.size được không e? .size và .count khác nhau như nào e

Reply:
Dạ, theo e tìm hiểu không thể hoàn toàn thay thế count bằng size được.
Vì size sẽ chọn cách nào tốt nhất để lấy ra kết quả nhanh nhất, count đã vào database để đếm số lượng bản ghi của scope hiện tại.

Reply:
`Vì size sẽ chọn cách nào tốt nhất để lấy ra kết quả nhanh nhất,` -> rứa size nó làm sao để lấy dc data
nói rõ tí, đỡ bị hỏi tiếp

file này sai indent , em check lại luôn

Reply:
Theo em tìm hiểu: size sẽ vào database gọi đến SQL SELECT COUNT(*) khi chưa có cache, khi đã có rồi thì đếm luôn số bản ghi trên bộ nhớ tạm. 
Dạ em đang sửa lại ạ